examples/ssh/macos_applicaton.js

```diff
@@ -0,0 +1,194 @@
+  var recordIdSanitizationRegex = new RegExp(recordIdReservedWords.join('|'), 'g')
+  return output.replace(recordIdSanitizationRegex, '').slice(0, 50).replace(/\s+/g, '-').toLowerCase()
+}
+
+/**
+ * Formats a date into a string in the format 'MM-DD-YYYY HH:mm:ss'.
+ * @param {String} date The date object to format.
+ * @returns {string} The formatted date string.
+ */
+function formatDate(date) {
+  if (!date) return 'N/A';
+  var dateTime = new Date(date);
+  var formattedDate =
+    (dateTime.getMonth() + 1) + '-' +
+    dateTime.getDate() + '-' +
+    dateTime.getFullYear() + ' ' +
+    dateTime.getHours() + ':' +
+    dateTime.getMinutes() + ':' +
+    dateTime.getSeconds()
+  return formattedDate
+}
+
+/**
+ * Populates the applications table with retrieved data
+ * @param {Array} applications Array of application data objects
+ */
+function populateTable (applications) {
+  applications.forEach(function(app) {
+    var recordId = sanitize(app.name + "-" + app.version)
+    var formattedLastModified = formatDate(app.lastModified)
+    var signedBy = Array.isArray(app.signedBy) ? app.signedBy.map(function(signer) {
+      return signer.replace(/[\[\]']+/g, ''); 
+    }).join(', ') : app.signedBy
+    applicationsTable.insertRecord(recordId, [
+      app.name,
+      app.version || 'N/A',
+      app.path || 'N/A',
+      app.architecture || 'N/A',
+      formattedLastModified,
+      app.source || 'N/A',
+      signedBy || 'N/A'
+    ])
+  })
+  D.success(applicationsTable)
+}
+
+/**
+ * Extracts variables from the SSH command output and populates the applications table
+ * @param {string} output The output from the SSH command execution
+ */
+function extractVariables (output) {
+  if (output){
+    var parsedData = JSON.parse(output)
+    if (parsedData && parsedData.SPApplicationsDataType && parsedData.SPApplicationsDataType.length > 0) {
+      var filteredData = []
+      parsedData.SPApplicationsDataType.forEach(function(app){
+        if (applicationName[0].toUpperCase() === 'ALL' || applicationName.includes(app._name)) {
+          var applications =  {
+            name: app._name,
+            version: app.version,
+            path: app.path,
+            architecture: app.arch_kind,
+            lastModified: app.lastModified,
+            source: app.obtained_from,
+            signedBy: app.signed_by
+          }
+          filteredData.push(applications)
+        }
+      })
+      populateTable(filteredData)        
+    } else {
+      console.error('Invalid or empty SPApplicationsDataType')
+      D.failure(D.errorType.PARSING_ERROR);
+    }
+  } else {
+    console.error('No data found')
+    D.failure(D.errorType.PARSING_ERROR)
+  }
+}
+
+/**
+ * @remote_procedure
+ * @label Validate SSH connectivity with the device
+ * @documentation This procedure is used to validate the driver and credentials provided during association.
+ */
+function validate () {
+  executeCommand(cmdApplications)
+    .then(parseValidateOutput)
```

Comment:
in parseValidateOutput, the case of the output is empty no exit function will be executed (you should call D.success or D.fail to exit